Cython/Compiler/Tests/TestGrammar.py

```diff
@@ -114,9 +114,9 @@ def test_invalid_number_literals(self):
                     # cython: language_level=3
                     ''' + code)
                 except CompileError as exc:
-                    assert code in [s.strip() for s in str(exc).splitlines()], str(exc)
+                    self.assertIn(code, [s.strip() for s in str(exc).splitlines()], str(exc))
                 else:
-                    assert False, "Invalid Cython code '%s' failed to raise an exception" % code
+                    self.assertTrue(False, "Invalid Cython code '%s' failed to raise an exception" % code)
```

Comment:
I don’t really like this `False, "Invalid Cython code '%s' failed to raise an exception" % code`, I would rather raise an exception ...

Reply:
This use case is covered by assertRaises 🙂

Reply:
@merwok Yes, it is. For this PR, I just replaced the `asserts` and did not really look at the rest of the code, but, probably, the change you proposed will still fit into the topic of the PR, so I will add it. Thanks for the comment.

Reply:
[resolved](https://github.com/cython/cython/pull/4340/commits/76042761eaffc36ff30b7df223d54344edd131dc)